# Alignments used in paper “Classification and monomer-by-monomer annotation of suprachromosomal family 1 alpha satellite higher-order repeats in hg38 human genome assembly”

## Alignments with master HOR monomers and consensus for MSA monomers:
1. S1-master-HOR-spec-aln-with-div-cons.fas — all monomers with HOR-specific alignment;
1. S1-master-HOR-auto-aln-with-div-cons-boxA.fas — separated A-type monomers auto aligned;
1. S1-master-HOR-auto-aln-with-div-cons-boxB.fas — separated B-type monomers auto aligned;
1. S1-master-HOR-manual-aln-171-with-div-cons-boxAB-sorted-typeCons.fas — combined A/B-type monomers with manual alignment to 171 bp, additional consensuses: SF class cons., A/B type cons., and J1m and J2m cons.

## Alignments with all HOR monomers and all MSA monomers:
1. S1-HOR-spec-aln-with-div-mons.fas — all monomers with HOR-specific alignment;
1. S1-HOR-auto-aln-with-div-mons-boxA.fas — separated A-type monomers auto aligned;
1. S1-HOR-auto-aln-with-div-mons-boxB.fas — separated B-type monomers auto aligned;
1. S1-HOR-manual-aln-171-with-div-mons-boxAB-sorted-typeCons.fas — combined A/B-type monomers with manual alignment to 171 bp, additional consensuses: SF class cons., A/B type cons., and J1m and J2m cons.
